skrifa/src/outline/hint.rs

```diff
@@ -4,11 +4,179 @@ use super::{
     cff,
     glyf::{self, FreeTypeScaler},
     pen::PathStyle,
-    AdjustedMetrics, DrawError, Hinting, LocationRef, NormalizedCoord, OutlineCollectionKind,
-    OutlineGlyph, OutlineGlyphCollection, OutlineKind, OutlinePen, Size,
+    AdjustedMetrics, DrawError, GlyphStyles, Hinting, LocationRef, NormalizedCoord,
+    OutlineCollectionKind, OutlineGlyph, OutlineGlyphCollection, OutlineKind, OutlinePen, Size,
 };
 use crate::alloc::{boxed::Box, vec::Vec};
 
+/// Configuration settings for a hinting instance.
+#[derive(Clone, Default, Debug)]
+pub struct HintingOptions {
+    /// Specifies the hinting engine to use.
+    ///
+    /// Defaults to [`Engine::AutoFallback`].
+    pub engine: Engine,
+    /// Defines the properties of the intended target of a hinted outline.
+    ///
+    /// Defaults to a target with [`SmoothMode::Normal`] which is equivalent
+    /// to `FT_RENDER_MODE_NORMAL` in FreeType.
+    pub target: Target,
+}
+
+impl From<Target> for HintingOptions {
+    fn from(value: Target) -> Self {
+        Self {
+            engine: Engine::AutoFallback,
+            target: value,
+        }
+    }
+}
+
+/// Specifies the backend to use when applying hints.
+#[derive(Clone, Default, Debug)]
+pub enum Engine {
+    /// The TrueType or PostScript interpreter.
+    Interpreter,
+    /// The automatic hinter that performs just-in-time adjustment of
+    /// outlines.
+    ///
+    /// Glyph styles can be precomputed per font and may be provided here
+    /// as an optimization to avoid recomputing them for each instance.
+    Auto(Option<GlyphStyles>),
+    /// Selects the engine based on the same rules that FreeType uses in
+    /// the default configuration.
+    ///
+    /// Specifically, PostScript fonts will always use the interpreter and
```

Comment:
```suggestion
    /// Specifically, PostScript fonts will always use the interpreter and
```
Do you mean CFF fonts, or PostScript fonts here? And by interpreter, do you mean the CFF hinting engine? Or do PS fonts have TrueType style hinting instructions?

Reply:
I’ve tried to be coherent in the docs wrt references so I use glyf/CFF/CFF2 when talking about tables and TrueType/PostScript when talking about the overall technologies.

This comment does refer to the CFF engine which is technically a PostScript interpreter :) If this is confusing, I’m open to suggestions for improvement.

Reply:
Thanks, then I understand what you mean. 

Perhaps something like this, to clarify that the two interpreters are different:
> Specifically, PostScript (CFF/CFF2) fonts will always use the PostScript interpreter and PostScript hinting engine. TrueType fonts will use the interpreter for TrueType instructions if one of the `fpgm` or `prep` tables is non-empty, falling back to the automatic hinter otherwise.

Reply:
That revision looks good to me, thanks!

Reply:
Also done